setup.py

```diff
@@ -58,11 +58,13 @@ def main():
                 'futures==3.1.1',
                 'functools32>=3.2,<=3.99'
             ],
-            'pubnub': ['pubnub>=4.0.0,<=4.99']
+            'pubnub': ['pubnub>=4.0.0,<=4.99'],
+            'joblib': ['joblib>=0.11.0,<=0.11.99'],
```

Comment:
i don't have a strong opinion on this particular case because i think pubnub and joblib are pretty lightweight (and pure python?), but keeping soft dependencies doesn't seem like a bad idea.  it could be very annoying to some user if they have trouble installing our software because of a conflict having to do with functionality they don't use.